src/datatypes/epgsql_codec_json.erl

```diff
@@ -17,18 +17,47 @@
 
 -define(JSONB_VERSION_1, 1).
 
-%% TODO: JSON encode/decode `fun Mod:Name/1` / `{Mod, Name}` as option.
-%% Shall not pass `fun(_) -> .. end`, because of hot code upgrade problems.
-init(_, _) -> [].
+%% JsonMod shall either be a module with encode/1 and decode/1, where:
+%%
+%% - encode/1 expects erlang-formatted JSON, e.g. a map or a proplist,
+%%   and returns a valid JSON binary string.
+%% - decode/1 expects a valid JSON binary string, and returns erlang-formatted JSON.
+%%
+%% or a tuple {Mod, {encode, Opts}, {decode, Opts}} where Mod implements encode/2 and
+%% decode/2 with the same interface semantics as encode/1 and decode/1, while
+%% accepting Opts as options to pass into the encode/2 and decode/2 functions.
+%%
+%% Tip: if you require special behavior that a third-party JSON module doesn't provide
+%%      for you, or which doesn't strictly meet this interface requirement, wrap the
+%%      third-party JSON module in a new module that implements the required interface,
+%%      and pass in _that_ as JsonMod.
+init(JsonMod, _) ->
+    JsonMod.
 
 names() ->
     [json, jsonb].
 
+encode(ErlJson, json, JsonMod) when is_atom(JsonMod) ->
+    JsonMod:encode(ErlJson);
```

Comment:
I would expect complaints from elvis code linter about variable-module-calls from non-behaviour module. I think it makes sense to make `epgsql_codec_json` a behaviour module by defining 4 `-callback` attributes.

Reply:
Would that require the `JsonMod` implementing those callbacks to specify the `-behavior` directive too though? If so that'd effectively require every user of this feature to create their own mini wrapper module around their favorite JSON library to implement the callbacks.

Reply:
no, it's NOT required for callback to actually have `-behavior` attribute